src/main/java/org/passay/DigestDictionaryRule.java

```diff
@@ -0,0 +1,69 @@
+/* See LICENSE for licensing and NOTICE for copyright. */
+package org.passay;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import org.cryptacular.bean.HashBean;
+
+/**
+ * Rule for determining if a password matches a digested password that is stored in a dictionary.
+ *
+ * @author  Middleware Services
+ */
+public class DigestDictionaryRule extends AbstractDictionaryRule
+{
+
+  /** Error code for matching dictionary word. */
+  public static final String ERROR_CODE = "ILLEGAL_DIGEST_WORD";
```

Comment:
Why would you want to use a different error code for this implementation? Isn't the semantic sense exactly the same as non-digest implementations?

Reply:
The message fundamentally has to be different. So this allows a deployer to use both rules at the same time. Maybe it's far fetched for someone to use two different types of dictionaries.

Reply:
Ok that makes sense. Probably as edge case as you noted, but it's clearly justifiable.